src/LiveComponent/src/Metadata/LiveComponentMetadataFactory.php

```diff
@@ -78,9 +84,17 @@ public function createLivePropMetadata(string $className, string $propertyName,
             }
         }
 
-        $type = $property->getType();
-        if ($type instanceof \ReflectionUnionType || $type instanceof \ReflectionIntersectionType) {
-            throw new \LogicException(sprintf('Union or intersection types are not supported for LiveProps. You may want to change the type of property %s in %s.', $property->getName(), $property->getDeclaringClass()->getName()));
+        if (null === $type && null === $collectionValueType && isset($infoTypes[0])) {
+            $typeClass = $infoTypes[0]->getClassName();
+
+            return new LivePropMetadata(
+                $property->getName(),
+                $liveProp,
+                $typeClass,
```

Comment:
If the phpdoc is, for example, an `int`, this argument should be passed `int`, not null. We need to check if the `Type` (i.e.`$infoTypes[0]`) is has a `getBuiltinType()` of `BUILTIN_TYPE_OBJECT`, then either pass `->getClassName()` here or `getBuiltinType()`. 

Also, I don't love choosing the `[0]` index. Does that mean that `@var null|ChildDTO` would fail? Is the `0` index the `null` type in that case? Could we loop over all of them to try to grab the "strongest" (non-null) type we can find?

Reply:
I've tested the property info with `null|*`case and then it always renders the last parts.
For example:
```
// @var null|int
array:1 [
  0 => Symfony\Component\PropertyInfo\Type {#48476
    -builtinType: "int"
    -nullable: true
    -class: null
    -collection: false
    -collectionKeyType: []
    -collectionValueType: []
  }
]

// @var null|\stdClass
array:1 [
  0 => Symfony\Component\PropertyInfo\Type {#48476
    -builtinType: "object"
    -nullable: true
    -class: "stdClass"
    -collection: false
    -collectionKeyType: []
    -collectionValueType: []
  }
]

// @var null|\stdClass|customobject
array:2 [
  0 => Symfony\Component\PropertyInfo\Type {#36519
    -builtinType: "object"
    -nullable: true
    -class: "stdClass"
    -collection: false
    -collectionKeyType: []
    -collectionValueType: []
  }
  1 => Symfony\Component\PropertyInfo\Type {#48476
    -builtinType: "object"
    -nullable: false
    -class: "customobject"
    -collection: false
    -collectionKeyType: []
    -collectionValueType: []
  }
]
```

just strange thing that in the last example first type is nullable and second is not (and this is always, if you use 3 or more union types then only first one is nullable)